src/composables/useRaycaster/index.ts

```diff
@@ -43,10 +48,10 @@ export const useRaycaster = (
 
     raycaster.value.setFromCamera(new Vector2(x, y), camera.value)
 
-    return raycaster.value.intersectObjects(objects.value, false)
+    return raycaster.value.intersectObjects(objects.value, true)
```

Comment:
If I understand correctly, this will test against all children of the scene and recurse down through all children.

If that's right, is that worth optimizing in this PR? Or ever?

----

For context, it looks like R3F does [some optimization here to keep some eventless objects](https://github.com/pmndrs/react-three-fiber/blob/bc0d4023a256d2cb0e04450d0803ceb3aa3928f4/packages/fiber/tests/core/renderer.test.tsx#L392) from being tested for intersections.

Reply:
Sounds good! 

It sounds like there's plenty to think about and discuss with the others. I don't want to keep these events out of Tres. So maybe we open another PR when this PR is merged.

> Only include node's with events and their direct descendants in the hit test. Being a custom renderer gives us this flexibility to do this on scene insertion.

Yeah, something like that is what I had in mind as well. We'd have to watch event updates as well.

Reply:
After talking with @alvarosabu and @Tinoooo it looks like we have enough time before the V4 release for me to take a stab at optimizing this some more

I was able to confirm `intersectObjects` is being called multiple times per frame (and even per event).
I've simplified it's usage. I'm seeing about half the cpu % in my test scenes.

Reply:
I think we could add a flag to the localstate `__tres` of the instances marking if we expect it to have events to optimize the traversing of the scene. See recently merged #522 

Meshes and groups:

```
instance.__tres.events = true
```

Dunno, materials and geometries:

```
instance.__tres.events = false
```

Reply:
Ahh good idea. The localstate `__tres` will be very helpful

Thank you for calling it out!

Reply:
I've added the EventManager onto Tres's Context

Let me know if that was what you had in mind